engine/classes/ElggFile.php

```diff
@@ -80,6 +80,18 @@ public function getFilenameOnFilestore() {
 	}
 
 	/**
+	 * Returns path relative to the filestore (data directory) root
+	 *
```

Comment:
Perhaps renaming to `getPathFromFilestoreRoot` and adding `getPathToFilestore` would make it more usable. Plugins can then compare filestore path to dataroot path.